### ISSUE TITLE
Bridge fails on LaunchRequest

### ISSUE THREAD

**Issue by devedse** [CatalystCode/alexa-bridge#7]
I tried setting up the AlexaBridge locally but I ran into some errors:

![image](https://user-images.githubusercontent.com/2350015/40781100-ff8fc22a-64da-11e8-91c3-541f0cef3cf9.png)

For some reason whenever I type "start bridge" in the amazon alexa test suite a LaunchRequest message is being sent to the bridge. However the bridge doesn't seem to be able to handle this.

I want to create a conversational bot using the Bot Framework.

**Comment by Endeavour-BRM** [comment]
I see there is no condition in the bridge script to handle Launch Request. Does this require fix.

**Comment by devedse** [comment]
I did do some fixes in my own fork of this repository. I'm not sure which of the changes we want to include:
https://github.com/devedse/alexa-bridge/commit/2d6cad8283735208d2460598b2d5b04dcc38403a

Furthermore it might be good to look at pointing the Azdeploy script to CatalystCode instead of daltskin:
https://github.com/devedse/alexa-bridge/commit/3ba927dbf31901aec7d15cf2a0bae2965f8ca856

**Comment by Endeavour-BRM** [comment]
@devedse Thanks. That helped. Request is going to bot framework. But get "timeout from alexa" and session ends before bot responds. Even with basic echo bot hosted in azure facing the same issue. I can see in the console after the session is ended message is received by the bridge from bot.

Even the bridge is hosted in cloud to avoid network latency.

Any ides how can I resolve this.

Please find the console output from the alexa bridge

Alexa utterance: jenkins
alexaIntent response time: 11
connector response time: 392
Subscribe event: Fdcra5NQUiHJc3wopizAiL|0000008 from: tcgbotwebapp message: You said jenkins
Alexa says:
SessionEndedRequest
closing the session
Alexa message out: You said jenkins.
Alexa Response Duration: 30001

**Comment by devedse** [comment]
Did you enable the direct channel and configure that?
Also the BotId is the name of the bot, not a guid.

**Comment by Endeavour-BRM** [comment]
@devedse yes. Direct channel is enabled. And you can see after the session closed because of timeout, bridge receives the message from bot "You said jenkins" and Alexa message out is constructed. But session is already closed. So message is not received by alexa.

**Comment by devedse** [comment]
What we did was rebuild this in C# using the AlexaSkillsKit Nuget package as this implementation is missing verification anyway. The effort was about 1 day of work.

**Comment by GhostAtom** [comment]
@devedse did you pushed the c# version to github?

**Comment by devedse** [comment]
@GhostAtom I don't have the IP of the c# version, sorry :/
